docs/oauth-setup.md

```diff
@@ -0,0 +1,143 @@
+# Using an external OAuth provider with S.H.I.E.L.D
+
+## Github
+
```

Comment:
Explain how Github Authentication works, that it uses OAuth2, that it can handle on-premise Github enterpries, etc.  As an operator, I'm going to be wondering "Why would I want to use this provider?"